tests/test_decorators.py

```diff
@@ -0,0 +1,83 @@
+from unittest.mock import Mock, patch
+
+import pytest
+
+from tululbot.decorators import CommandDispatcher, CommandNotFoundError
+
+
+def test_no_matching_command():
+    dispatcher = CommandDispatcher()
+    mock_callback = Mock()
+
+    dispatcher.add_command(r'^/hello$', mock_callback)
+    with pytest.raises(CommandNotFoundError):
+        dispatcher.run_command('/world')
+
+    assert not mock_callback.called
+
+
+def test_run_command_returns_the_same_value_as_callback():
+    dispatcher = CommandDispatcher()
+    mock_callback = Mock()
+    mock_callback.return_value = 'foo bar'
+
+    dispatcher.add_command(r'^/hello$', mock_callback)
+    rv = dispatcher.run_command('/hello')
+    assert rv == 'foo bar'
+
+
+def test_callback_is_called_with_regex_params():
+    dispatcher = CommandDispatcher()
+    mock_callback = Mock()
+
+    dispatcher.add_command(r'^/hello (\w+) (\w+)$', mock_callback)
+    dispatcher.run_command('/hello foo bar')
+    mock_callback.assert_called_once_with('foo', 'bar')
+
+
+def test_multiple_commands():
+    dispatcher = CommandDispatcher()
+    mock_callback1 = Mock()
+    mock_callback2 = Mock()
+
+    dispatcher.add_command(r'^/hello$', mock_callback1)
+    dispatcher.add_command(r'^/world$', mock_callback2)
+    dispatcher.run_command('/world')
+    assert not mock_callback1.called
+    assert mock_callback2.called
```

Comment:
I disagree. Asserting the invocation argument has been done by the previous test. No need to do that again. This test basically asserts that when multiple commands are registered and only one of the pattern matches, only the callback corresponding to the matching pattern is called. Here we don't care how or how many it is called; as long as it is called and the others aren't, the test passes.